### DISCUSSION TITLE
Roadmap/Strategy to support latest angular versions

### DISCUSSION THREAD

**Discussion by hjbolide** [vmware-clarity/ng-clarity#1492] [Q&A]
Hi,

First of all I want to thank you all for this amazing library and comprehensive style guides.

Second I want to apologies upfront for asking rookie question. I'm just wondering if there's any roadmap for supporting latest angular versions? I couldn't find the new version policies on clarity design website and no one seem to have asked about this.

Is there a branch we could actively tap into to use the latest development supporting new angular version?

Can someone please provide some insight?

Thank you very much,
Charles

**Comment by dangrussell** [comment]
Hello! I'm also a fan and consumer of Clarity. I would also like to know about the roadmap/strategy to support latest Angular versions. I've been following the progress the work being done toward Angular 18 compatibility and waiting patiently.

On the Clarity website, support for various Angular versions is outlined here: https://clarity.design/pages/support-policies As of August 15, 2024 (a week ago at the time of this comment), Clarity v17 is the only version actively supported.

This pull request to bring Clarity up to supporting Angular 18 was initiated on June 13, 2024: https://github.com/vmware-clarity/ng-clarity/pull/1440  By July 27th, 2024 the PR was approved and merged into the main branch.

Currently, the code that should make Clarity compatible with Angular 18 is in the main branch awaiting the next version (v18, presumably) to be cut and released. This answer says Angular 18 compatibility could even be included in the next minor version of Clarity (v17.3.0): https://github.com/vmware-clarity/ng-clarity/discussions/1422#discussioncomment-10168251

Clarity just happens to be the last package in the stack I'm using that does not yet support Angular 18, which is keeping me at Angular 17 for now. I absolutely do not want to be a pain about this -- I understand that real hard work is being done on this free project by real people.  I this is a great project and any more insight would definitely be welcomed.

Thank you!

**Reply by repolevedavaj** [reply]
I agree with @dangrussell. We already did the upgrade to Angular 18 and it worked without issues. But since then we are suffering from npm telling us that we have a dependency version conflict 😅. So it would be very nice to know when we can expect a release containing the Angular 18 compatibility. 

And just as a note: Angular 19 is expected to be released in the week of [19th of November 2024](https://angular.dev/reference/releases#release-schedule). Angular 18 has been released on 22th of May 2024.
